### DISCUSSION TITLE
Electron wrapper for offline gameplay...and other features I would like.

### DISCUSSION THREAD

**Discussion by cheez26** [okaybenji/text-engine#42] [Ideas]
Hi. So, I love the potential of this engine and I got questions and maybe ideas.

First, the ideas
- a super simple, ultra-thin wrapper for potential offline text adventures
- some kinda basic audio implementation, whether for modern or retro sounds for more immersive gameplay maybe...?

Anyhow, I might potentially use this anyway.

I do have questions about fonts and CSS: how do I implement a font into the games with the stylesheets? Would I have to make a new CSS file from scratch and go from there.

Thanks.

Chelsea, aka Cheez26

**Comment by okaybenji** [comment]
Hey, welcome! text-engine should play well with most other libraries you might want to use. I personally don't want _this_ library to get too opinionated about where and how it's used, so rather than adding Electron, audio implementation, etc. to this engine my recommendation would be to create a new repo for your game and pull text-engine, Electron, etc. into that.

Or alternatively, you could create your own repo for an opinionated text adventure stack that includes all that stuff and make it available to others for their games. I think that would be great! And I'd be happy to help if you have questions about that approach.

If you're looking for a good HTML audio solution, I would recommend [Howler.js](https://howlerjs.com). That will give you the ability to play audio files like wav, ogg, mp3, webm, etc. (if that's the vibe you're going for).

There are a couple of approaches you can take with the fonts. The easiest thing to do would be to import a Google Font. You can see an example of this in this project's index.html file:

`<link href="https://fonts.googleapis.com/css2?family=Nanum+Gothic+Coding:wght@700&family=Work+Sans&display=swap" rel="stylesheet">`

You can search the Google Fonts website for a font you like, and it will generate code like the above which you can use in your game. You will need to update the CSS to use your font. Take a look at styles/modern.css for an example of this.

If you don't want to use a Google Font, you can also just put a font file in the fonts directory and import that in your CSS. Look at styles/retro.css for an example of that.

Best of luck!

**Reply by cheez26** [reply]
Thanks for the response. I understand you want this to be as barebones as it can be while still acting as a template for beginning JS programmers, so take your time on the next update for any new feature you have already planned.
